### PR TITLE
Fix custom unit declaration

### PR DESCRIPTION
**Pull request by vdwees**
failing test case for custom units

### CODE REVIEW

**Comment by jackvreeken** [comment]
Thanks, but I cannot merge this as it breaks the CI testing. Could you either 

1. Turn it into a "WIP: fix custom unit declaration" and commit the actual fix as well, or
2. Disable the test with @unittest.skip, and add an issue that we still need to fix this test. I don't think it'd be that hard of an issue, so sticking a "good first issue" label on it seems like a good idea to me.

**Comment by jackvreeken** [comment]
@vdwees > I am not entirely happy about doing `ast.symbol.ATTRIBUTES[:-1]` everywhere, but then again.. I'm also not happy about the way `variable_metadata` caching is currently implemented, and the work required to fix it (which would be a separate issue imo).

**Comment by jbaayen** [comment]
Parsing a simple hydraulic model using RTC-Tools Channel Flow still results in the following:

```
File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/rtctools/optimization/modelica_mixin.py", line 48, in __init__
    self.__pymoca_model = pymoca.backends.casadi.api.transfer_model(kwargs['model_folder'], model_name, self.compiler_options())
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pymoca/backends/casadi/api.py", line 302, in transfer_model
    model = _compile_model(model_folder, model_name, compiler_options)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pymoca/backends/casadi/api.py", line 94, in _compile_model
    model = generator.generate(tree, model_name, compiler_options)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pymoca/backends/casadi/generator.py", line 756, in generate
    flat_tree = flatten(ast_tree, component_ref)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pymoca/tree.py", line 1001, in flatten
    flat_class = flatten_class(orig_class)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pymoca/tree.py", line 793, in flatten_class
    instance_tree = build_instance_tree(orig_class, parent=orig_class.parent)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pymoca/tree.py", line 398, in build_instance_tree
    sym.type = build_instance_tree(c, sym.class_modification, c.parent)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pymoca/tree.py", line 384, in build_instance_tree
    arg.value.component = arg.value.component.child[0]
IndexError: list index out of range
```

**Comment by jackvreeken** [comment]
That's not what this PR is supposed to fix. Test with https://github.com/pymoca/pymoca/pull/107.

File: src/pymoca/backends/casadi/api.py [hunk 1]
```diff
@@ -250,7 +250,7 @@ def _load_model(model_folder: str, model_name: str, compiler_options: Dict[str,
         for key in variables_with_metadata:
             for i, d in enumerate(db[key]):
                 variable = variable_dict[d['name']]
-                for j, tmp in enumerate(ast.Symbol.ATTRIBUTES):
+                for j, tmp in enumerate(ast.Symbol.ATTRIBUTES[:-1]):
```
[the remark this answers is not in the document]
Comment (jackvreeken) [reply src/pymoca/backends/casadi/api.py]: Agreed, updated accordingly.

**Pull request merged**